### ISSUE TITLE
reduce code duplication

### ISSUE THREAD

**Issue by jvoigtlaender** [fmidue/modelling-tasks#188]
Look at the clones reported by `jscpd`, e.g., here: https://github.com/jvoigtlaender/modelling-tasks-internal/actions/runs/3259787606/jobs/5352874648, and check which of them can be reasonably avoided by abstraction or by putting joined code/definitions into separate modules.

**Comment by nimec01** [comment]
Is this still relevant? `jscpd` does not report any duplicate code anymore.

**Comment by jvoigtlaender** [comment]
> `jscpd` does not report any duplicate code anymore.

Yeah, but the threshold of acceptable duplicates was increased to 4 percent, while it was 0 percent when this issue was opened. (And also `"minLines": 15` was added at some point.)

https://github.com/jvoigtlaender/modelling-tasks-internal/commit/1fe4c265a8548f1a13032414573a44a476fe81fe

https://github.com/jvoigtlaender/modelling-tasks-internal/commit/380960cdada31e0b17d3d5187a8f102a859fafd2

So, I guess, a reasonable approach would be to at least for estimation reducing those parameters somewhat, then see what comes up in terms of clones reported, then decide.

**Comment by marcellussiegburg** [comment]
Also, there are some duplicates still reported:
See e.g. starting at this line: https://github.com/jvoigtlaender/modelling-tasks-internal/actions/runs/14195572119/job/39769816053#step:4:3431
